tests/unit_test/elife_article_xml/elife_article_xml_pipeline_test.py

```diff
@@ -0,0 +1,420 @@
+        input_data = [{
+            'key-1': 'value1'
+        }]
+        expected_data = [{
+            'key_1': 'value1'
+        }]
+        result = get_bq_compatible_json_dict(input_data)
+        assert result == expected_data
+
+    def test_should_remove_none_value(self):
+        input_data = [{
+            'key_1': None
+        }]
+        expected_data = []
+        result = get_bq_compatible_json_dict(input_data)
+        assert result == expected_data
+
+    def test_should_remove_empty_list_value(self):
+        input_data = [{
+            'key_1': []
+        }]
+        expected_data = []
+        result = get_bq_compatible_json_dict(input_data)
+        assert result == expected_data
+
+
+class TestGetArticleJsonDataFromXmlStringContent:
+    def test_should_return_empty_dict_if_the_xml_with_empty_root(self):
+        xml_string = '<article></article>'
+        return_value = get_article_json_data_from_xml_string_content(xml_string)
+        assert return_value == {}
+
+    def test_should_return_empty_dict_if_the_xml_has_empty_article_meta(self):
+        xml_string = '<article><front><article-meta></article-meta></front></article>'
+        return_value = get_article_json_data_from_xml_string_content(xml_string)
+        assert return_value == {}
+
+    def test_should_not_return_empty_dict_if_the_xml_has_no_article_meta_section(self):
+        xml_string = ''.join([
+            '<article article-type="article_type_1" xmlns:xlink="http://www.w3.org/1999/xlink">',
+            '<front>',
+            '<other2>"other2"</other2>',
+            '</front>',
+            '</article>'
+        ])
+        return_value = get_article_json_data_from_xml_string_content(xml_string)
+        assert return_value == {}
+
+    def test_should_not_return_empty_dict_if_the_xml_has_no_front_section(self):
+        xml_string = ''.join([
+            '<article article-type="article_type_1" xmlns:xlink="http://www.w3.org/1999/xlink">',
+            '<other2>"other2"</other2>',
+            '</article>'
+        ])
+        return_value = get_article_json_data_from_xml_string_content(xml_string)
+        assert return_value == {}
+
+    def test_should_return_dict_with_related_article_and_article_if_if_they_exist(self):
+        xml_string = ''.join([
+            '<article article-type="article_type_1" xmlns:xlink="http://www.w3.org/1999/xlink">',
+            '<front><article-meta>',
+            '<article-id pub-id-type="publisher-id">publisher_id_1</article-id>',
+            '<article-id pub-id-type="doi">doi_1</article-id>',
+            '<related-article ext-link-type="doi" id="ra1" ',
+            'related-article-type="related_article_type_1" xlink:href="related_article_doi_1"/>',
+            '</article-meta></front>',
+            '</article>'
+        ])
+
+        return_value = get_article_json_data_from_xml_string_content(xml_string)
+        assert return_value == {
```

Comment:
I believe the structure has changed to what we had in the test yesterday. It no longer contains the `article`, `front` and `article-meta`.
i.e. if we wanted to include other elements outside `article-meta`, then the structure would likely need to change.